packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -61,4 +56,96 @@ export function watchTransactionForConfirmations<
 			}
 		})() as unknown;
 	}, web3Context.transactionReceiptPollingInterval ?? web3Context.transactionPollingInterval);
+};
+
+const waitBySubscription = ({
+	web3Context,
+	transactionReceipt,
+	transactionPromiEvent,
+	returnFormat,
+}: WaitProps) => {
+	setImmediate(() => {
+		web3Context.subscriptionManager
+			?.subscribe('newHeads')
+			.then((subscription: NewHeadsSubscription) => {
+				subscription.on('data', async (data: BlockOutput) => {
+					if (!data?.number) {
+						return;
+					}
+					const confirmationNumber =
+						BigInt(data.number) - BigInt(transactionReceipt.blockNumber);
+					if (confirmationNumber >= web3Context.transactionConfirmationBlocks) {
+						transactionPromiEvent.emit('confirmation', {
+							confirmationNumber: format(
+								{ eth: 'uint' },
+								confirmationNumber,
+								returnFormat,
+							),
+							receipt: transactionReceipt,
+							latestBlockHash: format(
+								{ eth: 'bytes32' },
+								data.parentHash as HexString32Bytes,
+								returnFormat,
+							),
+						});
+						await subscription.unsubscribe();
+					}
+				});
+				subscription.on('error', () => {
+					waitByPolling({
+						web3Context,
+						transactionReceipt,
+						transactionPromiEvent,
+						returnFormat,
+					});
+				});
+			})
+			.catch(() => {
+				waitByPolling({
+					web3Context,
+					transactionReceipt,
+					transactionPromiEvent,
+					returnFormat,
+				});
```

Comment:
I think this should be replace by [this error](https://github.com/ChainSafe/web3.js/blob/d805cf9f7d382f19739018810c30d94a6f2f749e/packages/web3-core-method/src/index.js#L545)